Add support for Tiny Tiny Rss

razcampagne opened the pull request:
Hi,

As promised, here is a commit adding support for Tiny Tiny Rss.
All features of a typical fetcher subclass are implemented but some more advanced features would require more involved changes in the qml code and I didn't want to disturd the UI too much just for this backend.

mkiol commented:
It looks amazing!!! Please give me few days to setup tts instance and check how it works. 

In the meantime, if it's not a problem, could you please add your self as an author in the files you created?

razcampagne commented:
Thank you :)
It's not perfect but it should be working. I tried supporting qt4 but I couldn't test it. I'm not even sure it is still relevant nowadays but well…
I added my name in the files. Tell me if it's okay with you.

razcampagne commented:
I might have pushed the Pull Request button a bit too soon.
I fixed several bugs and removed the slow/liked viewMode for ttrss backend. There is a published view mode in ttrss I would like to add but I'd need to have a dedicated icon for this but I'm not sure how to proceed. How did you make the icons?

mkiol commented:
Sorry for the delay. I'm a little a busy right now. I'm going to review and merge your change during weekend, promise.

razcampagne commented:
No worries, take you time.
Actually I'm still finding and fixing bugs as I'm using it.

mkiol commented:
Thank you for your contribution. It is fabulous! I'm switching from oldreader to tt-rss :-)
> I'd need to have a dedicated icon for this but I'm not sure how to proceed. How did you make the icons?

In fdc282a6b7876c9e4e6e62896c4bd491df0a0c61 I've submitted raw SVG files for all icons I've made for Kaktus. You can use them as a reference. The [res](https://github.com/mkiol/kaktus/tree/master/sailfish/res) directory also contains small script `gen.sh`. It generates PNG files in different sizes. To make it work Inkscape and Python must be installed in your system.
